sim-lib/src/lib.rs

```diff
@@ -674,12 +709,22 @@ impl Simulation {
 
             // Generate a consumer for the receiving end of the channel. It takes the event receiver that it'll pull
             // events from and the results sender to report the events it has triggered for further monitoring.
-            tasks.spawn(consume_events(
-                node.clone(),
-                receiver,
-                output_sender.clone(),
-                self.shutdown_trigger.clone(),
-            ));
+            // ce: consume event
+            let ce_listener = self.shutdown_listener.clone();
+            let ce_shutdown = self.shutdown_trigger.clone();
+            let ce_output_sender = output_sender.clone();
+            let ce_node = node.clone();
+            tasks.spawn(async move { 
+                if let Err(e) = consume_events(
```

Comment:
nice to have but non-blocking: a lot of these functions that we spawn in tasks have a "function_name starting" debug log - would be nice to move that up to this level as well. 

Could also update the `if let` to a `match` and log when it exits without error (since we have that in the function as well).